src/signature.c

```diff
@@ -414,8 +414,9 @@ int verify_callback(int ok, X509_STORE_CTX *stor)
  * returns: true if signature verification succeeded, false if verification
  * failed, or the signature download failed
  */
-bool download_and_verify_signature(const char *data_url, const char *data_filename)
+bool download_and_verify_signature(const char *data_url, const char *data_filename, int version, bool mix_exists)
```

Comment:
You could simplify this change by only adding one extra parameter instead of two. Maybe

```
bool download_and_verify_signature(const char *data_url, const char *data_filename, const char *mixer_sig)
```

The caller would initialize mixer_sig to NULL for the non-mix case, or set the full path for the mix case. Then the conditional code below can check if mixer_sig is NULL or non-NULL.